### PR TITLE
Cleanup API

### PR DESCRIPTION
**Pull request by karnkaul**
- Ensure `EnumArray`s are fully initialized.
- Refactor `utils::Size` to `TEnumValue`.
- Add `utils::Rate`.
- Rename `SampleMeta` to `Metadata`, move to its own header, include in `sound.hpp`.
- Unify Sound/Music API.
- Cleanup examples.
- Upgrade (and bugfix) `ktl::expected`, remove `Outcome`.

### CODE REVIEW

File: include/capo/pcm.hpp [hunk 1]
```diff
@@ -35,9 +21,11 @@ struct PCM {
 
 	static constexpr std::size_t max_channels_v = 2;
 
-	SampleMeta meta;
+	Metadata meta;
 	std::vector<Sample> samples;
-	utils::Size size{};
+	std::size_t bytes{};
```
Comment (alexdevteam) [inline include/capo/pcm.hpp:26]: I don't see anything wrong with keeping a Size object here instead of a std::size_t. It's more specific.

**Review by alexdevteam — APPROVED** [review]

File: include/capo/pcm.hpp (hunk 1 above)
Comment (karnkaul) [reply include/capo/pcm.hpp:26]: It stores the result as a float + unit, which may cause imprecision. It's meant for human friendly output, not really for precise storage.

Comment (alexdevteam) [reply include/capo/pcm.hpp:26]: What about having size store bytes as size_t instead of float?

Comment (karnkaul) [reply include/capo/pcm.hpp:26]: Then the original purpose - of representing the value in the nearest thousandth unit - becomes less precise: numbers beyond decimal point are lost. 🤓

**Pull request merged**